src/utils/sendSearchRequest.ts

```diff
@@ -1,158 +1,213 @@
-import { SummaryLanguage } from "../types";
+import {ChatQueryBody, DEFAULT_DOMAIN, ChatQueryRequestHeaders, SummaryLanguage} from "../types";
 
+type GenerationConfig = {
+    promptName?: string;
+    maxUsedSearchResults?: number;
+    promptText?: string;
+    maxResponseCharacters?: number;
+    responseLanguage?: SummaryLanguage;
+    modelParameters?: {
+        maxTokens: number;
+        temperature: number;
+        frequencyPenalty: number;
+        presencePenalty: number;
+    };
+    citations?: {
+        style: "none" | "numeric";
+    } | {
+        style: "html" | "markdown";
+        urlPattern: string;
+        textPattern: string;
+    };
+    enableFactualConsistencyScore?: boolean;
+};
 type Config = {
-  customerId: string;
-  corpusId: string;
-  endpoint: string;
-  apiKey: string;
-  filter: string;
-  queryValue?: string;
-  language?: SummaryLanguage;
-  summaryMode?: boolean;
-  rerank?: boolean;
-  rerankNumResults?: number;
-  rerankerId?: number;
-  rerankDiversityBias?: number;
-  hybridNumWords: number;
-  hybridLambdaShort?: number;
-  hybridLambdaLong?: number;
-  summaryNumResults?: number;
-  summaryNumSentences?: number;
-  summaryPromptName?: string;
-  enableFactualConsistencyScore?: boolean;
-  chat?: {
-    conversationId?: string;
-  };
+    customerId: string;
+    apiKey?: string;
+    authToken?: string;
+    domain?: string;
+    query: string;
+    corpusKey: string;
+    search: {
+        metadataFilter: string;
+        lexicalInterpolation?: number;
+        customDimensions?: Record<string, number>;
+        semantics?: "default" | "query" | "response";
+        offset: number;
+        limit?: number;
+        contextConfiguration?: {
+            charactersBefore?: number;
+            charactersAfter?: number;
+            sentencesBefore?: number;
+            sentencesAfter?: number;
+            startTag?: string;
+            endTag?: string;
+        };
+        reranker?: {
+            type: "none";
+        } | {
+            type: "customer_reranker";
+            rerankerId: string;
+        } | {
+            type: "mmr";
+            diversityBias: number;
+        };
+    };
+    generation?: GenerationConfig;
+    chat?: {
+        store?: boolean;
+        conversationId?: string;
+    };
+};
+
+const convertReranker = (reranker?: Config["search"]["reranker"]) => {
+    if (!reranker) return;
```

Comment:
We use 2 spaces for indentation - sorry I didn't catch it earlier, but would you mind reformatting please?